toolbox/db/db_get.m

```diff
@@ -46,12 +46,106 @@
 
 varargout = {};
 switch contextName
+%% ==== SUBJECT ====
+    % Usage : sSubject = db_get('Subject', SubjectIDs,       Fields, isRaw);
+    %         sSubject = db_get('Subject', SubjectFileNames, Fields, isRaw);
+    %         sSubject = db_get('Subject', CondQuery,        Fields, isRaw);
+    %         sSubject = db_get('Subject');
+    % If isRaw is set: force to return the real brainstormsubject description
+    % (ignoring whether it uses protocol's default anatomy or not)    
     case 'Subject'
-        iSubject = args{1};
-        varargout{1} = sql_query(sqlConn, 'select', 'subject', '*', struct('Id', iSubject));
+        % Default parameters
+        fields = '*';   
+        isRaw = 0;
+        templateStruct = db_template('Subject');
+
+        % Parse first parameter
+        if isempty(args)
+           ProtocolInfo = bst_get('ProtocolInfo');
+           iSubjects = ProtocolInfo.iSubject;
+        else
+           iSubjects = args{1};
+        end
+        % SubjectFileNames and CondQuery cases
+        if ischar(iSubjects)
+            iSubjects = {iSubjects};
+        elseif isstruct(iSubjects)
+            condQuery = args{1};           
+        end
+
+        % Parse Fields parameter
+        if length(args) > 1
+            fields = args{2};
+            if ischar(fields)
+                fields = {fields};
+            end
+            % Verify requested fields
+            if ~all(isfield(templateStruct, fields))
+                error('Invalid Fields requested in db_get()');
+            else
+                for i = 1 : length(fields)
+                    resultStruct.(fields{i}) = templateStruct.(fields{i});
+                end
+            end
+        else
+            resultStruct = templateStruct;
+        end
+                       
+        % isRaw parameter
+        if length(args) > 2
+            isRaw = args{3};
+        end
+        
+        % Input is SubjectIDs or SubjectFileNames
+        if ~isstruct(iSubjects)
+            nSubjects = length(iSubjects);
+            sSubjects = repmat(resultStruct, 1, nSubjects);
+            for i = 1:nSubjects
+                if iscell(iSubjects)
+                    condQuery.FileName = iSubjects{i};
+                else
+                    condQuery.Id = iSubjects(i);
+                end
+                result = sql_query(sqlConn, 'select', 'subject', fields, condQuery);
+                if isempty(result)
+                    error('Subject entry was not found in db_get()')
```

Comment:
Not a deal breaker but would be nice to get the missing subject in the error message.

Reply:
Added in 1120e3143f6839ed14eb94df8af74e60fdd5cd23